assets/src/scripts/components/hydrograph/audible.js

```diff
@@ -0,0 +1,119 @@
+const { scaleLinear } = require('d3-scale');
+const { select } = require('d3-selection');
+const memoize = require('fast-memoize');
+const { createSelector, createStructuredSelector } = require('reselect');
+
+const { yScaleSelector } = require('./scales');
+const { tsDatumSelector } = require('./tooltip');
+
+const { dispatch, link } = require('../../lib/redux');
+const { Actions } = require('../../store');
+
+
+// Higher tones get lower volume
+const volumeScale = scaleLinear().range([2, .3]);
+
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+const getAudioContext = memoize(function () {
+    return new AudioContext();
+});
+
+export const createSound = memoize(/* eslint-disable no-unused-vars */ tsKey => {
+    const audioCtx = getAudioContext();
+    const oscillator = audioCtx.createOscillator();
+    const gainNode = audioCtx.createGain();
+    const compressor = audioCtx.createDynamicsCompressor();
+
+    compressor.threshold.setValueAtTime(-50, audioCtx.currentTime);
+    compressor.knee.setValueAtTime(40, audioCtx.currentTime);
+    compressor.ratio.setValueAtTime(12, audioCtx.currentTime);
+    compressor.attack.setValueAtTime(0, audioCtx.currentTime);
+    compressor.release.setValueAtTime(0.25, audioCtx.currentTime);
+
+    // Connect the oscillator to the gainNode to modulate volume
+    oscillator.type = 'sine';
+    oscillator.connect(gainNode);
+
+    // Connect the gainNode to the compressor to address clipping
+    gainNode.connect(compressor);
+
+    // Connect the compressor to the output context
+    compressor.connect(audioCtx.destination);
+
+    // Start the oscillator
+    oscillator.start();
+
+    return {oscillator, gainNode, compressor};
+});
+
+export const updateSound = function ({enabled, points}) {
+    const audioCtx = getAudioContext();
+    for (const tsKey of Object.keys(points)) {
+        const point = points[tsKey];
+        const {oscillator, gainNode} = createSound(tsKey);
+        oscillator.frequency.setTargetAtTime(
+            enabled && point ? point : null,
+            audioCtx.currentTime,
+            .2
+        );
+        gainNode.gain.setTargetAtTime(
+            enabled && point ? volumeScale(point) : null,
+            audioCtx.currentTime,
+            .2
```

Comment:
As a general comment, I'm curious how you came up with this number and the other numbers?

Reply:
That one was trial and error... it smooths out the pitch changes so the transitions don't sound really harsh. The 80-1500 hz range was stolen from one of the D3 examples Jim found.